BUILDING.md

```diff
@@ -12,8 +12,127 @@ For building the manual, see the README in the [Manual repository](https://githu
 
 Windows
 -------
+ Visual studio version:
+ 1.) Download:
+   https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2017
+
+   Install: Tools for Visual Studio 2019
+     MSVC v142 
+     Windows 10 SDK
+     C++ Cmake Tools for Windows
+     MSVC v141 - VS2017 C++ x64/x86 build Tools
+
+ 2.) Download:
+   http://download.qt.io/official_releases/qt/5.9/5.9.7/qt-opensource-windows-x86-5.9.7.exe
+   Install 
+      MSVC 2017 64-bit
+ 3.) Compile (Console)
+
+   set Qt5_DIR=E:\Qt\Qt5.9.7\5.9.7\msvc2017_64\lib\cmake\Qt5
+   cd {DwarfTherapist Source Dir}
+   mkdir build
+   cd bild
+   cmake -G "Visual Studio 15 2017 Win64" ..\
+   MSBuild.exe DwarfTherapist.sln /p:Configuration=Release 
+
+ 4.) Prepare DT directory
+   Copy from E:\Qt\Qt5.9.7\5.9.7\msvc2017_64\bin
+   Qt5Core.dll
+   Qt5Gui.dll
+   Qt5Network.dll
+   Qt5Qml.dll
+   Qt5Widgets.dll to build directory 
+
+   Copy Memory_layouts to build directory -> should work 
+   (Exe starts but crashes, when loading a fortress - reason unknown so far)
+###############################################################
+Cygwin version:
+ 1.) Install cygwin and select these packages:
+   cmake
+   gcc-g++
+   git
+   make
+   libQt5Quick-devel
+   libQt5Svg-devel
+   xinit
+   xorg-server
+
+ 2.) Modify files:
+
+   Modify "E:\cygwin64\usr\include\qt5\QtGui\qwindowdefs_win.h" 
+   // Maybe this is a bug in QT5 sources?
+   ....
+   #ifndef _HRESULT_DEFINED
+   #ifdef __CYGWIN__         // # ADDED
+     typedef int HRESULT;    // # ADDED
+   #else                     // # ADDED
+     typedef long HRESULT;   
+   #endif                    // # ADDED
+   #endif
+   ...
+
+ 3.) Compile 
+   cd (DwarfTherpist Source Dir)
+   mkdir build
+   cd build
+   cmake ../
+   make
+ 4.) Prepare DT directory
+   Copy Memory_layouts to 
+   ~/.local/share/Dwarf\ Therapist
+ 5.) Run DT
+   startx
+   ./DwarfTherapist.exe
+
+does connect and load a fortress 
+###############################################################
+Cygwin cross compile for windows using mingw:
+ 1.) install cygwin like above and select these packages:
+   mingw64-x86_64-gcc-g++-7.4.0.1
+   mingw64-x86_64-qt5-base-5.9.4-1
+   mingw64-x86_64-qt5-quickcontrols-5.9.4-1
+   mingw64-x86_64-qt5-svg-5.9.4-1
+
+ 2.) use this file for cross compiling: Toolchain-mingw32.cmake
+
+ 3.) compile
+  cd (DwarfTherpist Source Dir)
+  mkdir build
+  cd build
+  cmake -DCMAKE_TOOLCHAIN_FILE=../Toolchain-mingw32.cmake ../
+  make
+
+ 4.) Prepare DT directory
+  copy these files from /usr/x86_64-w64-mingw32/sys-root/mingw
+  to the build directory
```

Comment:
MSYS2 Qt comes with a windeployqt too. although I am not sure it is working well. When I shipped mingw builds, I was using a script based on ntldd for finding the dlls to copy.

Reply:
Does this script recursively examine the DLL imports to find the need modules? I used "Depends" to find it. But it I already was thinking of some kind of script to avoid the manual steps.

Reply:
ntldd has a recursive option, but windeployqt should be preferred.